### PR TITLE
State management feature

### PR DESCRIPTION
**Pull request by patel1643**
Added state management library and set up basic skeleton for store

### CODE REVIEW

**Review by jamestang12 — CHANGES_REQUESTED** [review]

File: client/src/pages/Register.tsx [hunk 1]
```diff
@@ -37,64 +37,44 @@ type FormValues = {
...
-            email: data.email,
-            password: data.password
-        };
-        const URL = process.env.REACT_APP_DEV_SERVER_URL;
-        axios.post(URL + '/api/auth/register', userData)
+        setGeneralError(false);
+        registerUser(data)
             .then(res => {
+                reset();
                 console.log(res);
```
[the remark this answers is not in the document]
Comment (jamestang12) [reply client/src/pages/Register.tsx:61]: My bad I mean line 61

File: client/src/pages/Register.tsx [hunk 2]
```diff
@@ -37,64 +37,44 @@ type FormValues = {
...
                 onOpen();
-
+                setIsLoading(false);
             })
             .catch(err => {
                 console.log(err);
```
[the remark this answers is not in the document]
Comment (jamestang12) [reply client/src/pages/Register.tsx:66]: I mean line 66 here

File: client/src/pages/Login.tsx [hunk 3]
```diff
@@ -27,26 +30,61 @@ type FormValues = {
...
+                    store.setLoggedIn(true);
+                    setIsLoading(false);
+                    //Redirect to Dashboard page from here
+                }
+            })
+            .catch(err => {
+                setIsLoading(false);
+                if (err.response.data === 'Incorrect Password') {
+                    setError('password', {
+                        message: 'Wrong password!',
```
[the remark this answers is not in the document]
Comment (Mwpereira) [reply client/src/pages/Login.tsx:59]: The response errors should either use a msg coming from the server or have a msg based on the error code. Should not check for a string and provide a different msg.

File: client/src/pages/Login.tsx [hunk 4]
```diff
@@ -27,26 +30,61 @@ type FormValues = {
...
-        formState: { errors }
+        formState: { errors },
+        setError,
+        reset
     } = useForm<FormValues>();
+    const [isLoading, setIsLoading] = useState(false);
+
+    const handleSubmitForm = (data: FormValues) => {
+        login(data.email, data.password)
+            .then((res: any) => {
```
[the remark this answers is not in the document]
Comment (Mwpereira) [reply client/src/pages/Login.tsx:47]: Should try to use async

File: client/src/pages/Login.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (patel1643) [reply client/src/pages/Login.tsx:59]: gotcha

File: client/src/pages/Login.tsx [hunk 5]
```diff
@@ -27,26 +30,61 @@ type FormValues = {
...
+        return axios.post(URL + '/api/auth/login', userData);
+
+    };
 
     const minPassLength: number = 5;
 
     return (
         <Center p={'1rem'} height={'100vh'}>
             <ScaleFade initialScale={0.9} in>
+                <Text>Login Status: {loggedIn ? 'True => redirect to Dashboard' : 'False'}</Text>
```
[the remark this answers is not in the document]
Comment (patel1643) [reply client/src/pages/Login.tsx:87]: Ohh it's just there for testing at least for now

File: client/src/pages/Login.tsx [hunk 6]
```diff
@@ -27,26 +30,61 @@ type FormValues = {
...
+        formState: { errors },
+        setError,
+        reset
     } = useForm<FormValues>();
+    const [isLoading, setIsLoading] = useState(false);
+
+    const handleSubmitForm = (data: FormValues) => {
+        login(data.email, data.password)
+            .then((res: any) => {
+                if (res.data === 'Signed In') {
```
[the remark this answers is not in the document]
Comment (patel1643) [reply client/src/pages/Login.tsx:48]: yeah will do

File: client/src/pages/Login.tsx [hunk 7]
```diff
@@ -27,26 +30,61 @@ type FormValues = {
...
+                    reset();
+                    store.setLoggedIn(true);
+                    setIsLoading(false);
+                    //Redirect to Dashboard page from here
+                }
+            })
+            .catch(err => {
+                setIsLoading(false);
+                if (err.response.data === 'Incorrect Password') {
```
[the remark this answers is not in the document]
Comment (patel1643) [reply client/src/pages/Login.tsx:57]: @Mwpereira can you confirm this one?

File: client/src/pages/Login.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (patel1643) [reply client/src/pages/Login.tsx:47]: gotcha

File: client/src/pages/Login.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (patel1643) [reply client/src/pages/Login.tsx:59]: I'll update the code once I get an update from @Mwpereira

**Pull request closed without merging**